streams/src/main/java/org/apache/kafka/streams/kstream/internals/foreignkeyjoin/SubscriptionStoreReceiveProcessorSupplier.java

```diff
@@ -68,6 +68,8 @@ public void init(final ProcessorContext context) {
                     internalProcessorContext.metrics()
                 );
                 store = internalProcessorContext.getStateStore(storeBuilder);
+
+                keySchema.init(context);
```

Comment:
The shema object isn't a Serde, but it's similar in that it wraps nullable user-provided (de)serializers, so it also needs to allow defaulting.